plugins/module_utils/mso.py

```diff
@@ -294,96 +327,119 @@ def __init__(self, module):
 
         # debug output
         self.has_modified = False
-        self.filter_string = ''
+        self.filter_string = ""
         self.method = None
         self.path = None
         self.response = None
         self.status = None
         self.url = None
 
         if self.module._debug:
-            self.module.warn('Enable debug output because ANSIBLE_DEBUG was set.')
-            self.params['output_level'] = 'debug'
+            self.module.warn("Enable debug output because ANSIBLE_DEBUG was set.")
+            self.params["output_level"] = "debug"
 
-        if self.module._socket_path is None:
-            if self.params.get('use_ssl') is None:
-                self.params['use_ssl'] = True
-            if self.params.get('use_proxy') is None:
-                self.params['use_proxy'] = True
-            if self.params.get('validate_certs') is None:
-                self.params['validate_certs'] = True
+        # Ensure protocol is set
+        self.params["protocol"] = "https" if self.params.get("use_ssl", True) else "http"
 
-            # Ensure protocol is set
-            self.params['protocol'] = 'https' if self.params.get('use_ssl', True) else 'http'
+        if self.params.get("host") is None:
+            self.fail_json(msg="Parameter 'host' is required when not using the HTTP API connection plugin")
```

Comment:
You can't move this outside of "if self.module._socket_path is None:"  as this only applies to non-HTTPAPI plugin cases.